ctk_cli/module.py

```diff
@@ -172,6 +174,49 @@ def _parse(self, elementTree):
                 logger.warning("Element %r within %r not parsed" % (_tag(pnode), _tag(elementTree)))
 
 
+class CLIModuleList(object):
+    def __init__(self, path=None, env=None, stream=None, root=None):
+        """
+        Parse a list of CLI specifications from an XML document. This class can be
+        instantiated in several different modes:
+
+        1. Pass the ``path`` parameter that points to a CLI executable. This
+           mode will run the executable in a subprocess, passing "--xml"
+           and parsing the output.
+        2. Pass the ``path`` parameter representing the path to an XML
+           file on disk that contains the CLI description.
+        3. Pass a file-like object in the ``stream`` kwarg that contains
+           the XML document to be parsed.
+
+        :param path: The path on the filesystem of either the CLI to
+            be interrogated, or a CLI XML description document.
+        :param env: If using mode 1 described above, setting this parameter
+            causes "env=<value>" to be passed as an additional command line argument
+            when invoking the subprocess to describe the CLI.
+        :param stream: An open file-like object that will stream the CLI
+            XML description document.
+        :param root: The root of an existing ElementTree representing a CLIModuleList.
+        """
+        self.path = path
+
+        if path and isCLIExecutable(path):
+            elementTree = getXMLDescription(path, env=env)
+        elif path:
+            with open(path) as f:
+                elementTree = ET.parse(f)
+        elif stream is not None:
+            elementTree = ET.parse(stream)
+        elif root is None:
+            raise RuntimeError(
+                'You must pass a path, stream, or root when instantiating CLIModuleList.')
+
+        root = root or elementTree.getroot()
+        if root.tag == 'executables':
```

Comment:
There is a schema file here: https://raw.githubusercontent.com/commontk/CTK/master/Libs/CommandLineModules/Backend/LocalProcess/Resources/ctkCmdLineModuleProcess.xsd